src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -193,11 +195,14 @@ module.exports = function<T, P, I, TI, C>(
         commitNestedUnmounts(node);
         // After all the children have unmounted, it is now safe to remove the
         // node from the tree.
-        if (parent) {
+        if (parent && !ignoreHostNode) {
           removeChild(parent, node.stateNode);
         }
       } else {
-        commitUnmount(node);
+        // If ignoreTop is true, don't unmount the top fiber
+        if (!ignoreTop || node !== current) {
```

Comment:
Similar question here. Why are we sure this is a "top" node? It isn't quite clear to me.
Is there only one "top" node, or can this block run more than twice in a loop?
Is there a better name for `ignoreTop`? I don't understand which fiber exactly would be called a "top" node.